contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,76 @@
+ * Identification of resources: as mentioned before (see [Meaning Of REST](#meaningOfRest)), “resource” is an organization of information and the identifiers of the resources need to remain constant. An example to illustrate this constraint is [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier). 
+ * Manipulation of resources through representations: as mentioned before (see [Meaning Of REST](#meaningOfRest)), the “representation” of the resource at one point i.e. the representation of the state of the resource, is what the users see and modify, instead of the resource itself. 
+ * Self-descriptive messages: A message in a communication between the web components should contain all the information needed for the web components to understand its content. 
+ * Hypermedia as the engine of application state (HATEOAS): There should be hyperlinks embedded inside the representations given to the client, such that all the future actions that the client might take are within these representations. Hence, the client can interact with and navigate through the application without any prior knowledge of how to do so. Hence, the client and the server are more independent of each other. 
+
+### <a name="codeOnDemand"></a>Code On Demand (Optional)
+The server can send a code snippet to the client to let the client execute. One example of this is the Javascript code sent along with the webpage in HTML. This constraint extends the client functionality and reduces the workload of the server by reducing the number of features to be implemented at the server. However, it reduces the visibility of the interactions between the client and the server and makes it more difficult to monitor the interactions. Hence, this constraint might be disabled in some implementations in REST style.
+
+## Useful Resources
+* [Roy Fielding's paper which gave birth to REST](https://www.ics.uci.edu/~fielding/pubs/dissertation/fielding_dissertation.pdf)
+* [History of hypermedia and REST explained](https://ruben.verborgh.org/phd/hypermedia/)
+* Web intermediaries
+	* [Web intermediaries explained](http://www.almaden.ibm.com/cs/wbi/doc/Architecture.html)
+	* an important type of intermediaries: [proxy](https://en.wikipedia.org/wiki/Proxy_server#Filtering_of_encrypted_data)
+* Hypermedia as the engine of application state (HATEOAS)
+	* [A general wikipedia explanation](https://en.wikipedia.org/wiki/HATEOAS)
+	* [A more detailed explanation](https://ruben.verborgh.org/phd/hypermedia/#hypermedia-as-the-engine-of-application-state)
+* Code On Demand
+	* [a general wikipedia explanation](https://en.wikipedia.org/wiki/Code_on_demand)
+	* [a stackoverflow explanation](https://stackoverflow.com/questions/32094952/code-demand-constraint-for-restful-apis?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)
+* [Video explaning REST (easy to get the big picture but not very precise)](https://www.youtube.com/watch?v=YCcAE2SCQ6k)
```

Comment:
Good range of links although the "Web intermediaries explained" one has already been referenced elsewhere in the chapter.